File: android/VideoView/app/src/main/java/io/streamroot/dna/samples/videoview/PlayerActivity.java
package io.streamroot.dna.samples.videoview;

import android.net.Uri;
import android.os.Bundle;
import android.widget.MediaController;
import android.widget.Toast;
import android.widget.VideoView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Objects;

import io.streamroot.dna.core.DnaClient;
import io.streamroot.dna.core.log.LogLevel;
import io.streamroot.dna.utils.stats.StatsView;
import io.streamroot.dna.utils.stats.StreamStatsManager;

public final class PlayerActivity extends AppCompatActivity {
    private VideoView videoView;
    private StatsView streamrootDnaStatsView;

    private String mStreamUrl;

    private DnaClient streamrootDNA;
    private StreamStatsManager streamStatsManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_player);

        mStreamUrl = Objects.requireNonNull(getIntent().getExtras()).getString("streamUrl");

        videoView = findViewById(R.id.videoView);
        streamrootDnaStatsView = findViewById(R.id.streamrootDnaStatsView);
    }

    @Override
    protected void onResume() {
        super.onResume();

        initStreamroot();
        if (streamrootDNA != null) {
            videoView.setVideoPath(streamrootDNA.getManifestUrl().toString());
        } else {
            videoView.setVideoPath(mStreamUrl);
        }

        videoView.requestFocus();
        videoView.setMediaController(new MediaController(this));
        videoView.start();
    }

    @Override
    protected void onPause() {
        super.onPause();

        videoView.pause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        videoView.stopPlayback();
        stopStreamroot();
    }

    private void initStreamroot() {
        try {
            streamrootDNA = DnaClient.newBuilder()
                    .context(getApplicationContext())
                    .playerInteractor(new VideoViewPlayerInteractor(videoView))
                    .logLevel(LogLevel.OFF) // Default level: ERROR. Available: [VERBOSE, DEBUG, INFO, WARN, ERROR, OFF]
                    .start(Uri.parse(mStreamUrl));

            streamStatsManager = StreamStatsManager.newStatsManager(streamrootDNA, streamrootDnaStatsView);
        } catch (Exception e) {
            Toast.makeText(getApplicationContext(), e.getMessage(), Toast.LENGTH_LONG)
                    .show();
        }
    }

    private void stopStreamroot() {
        if (streamrootDNA != null) {
            streamrootDNA.close();
            streamrootDNA = null;
        }

        if (streamStatsManager != null) {
            streamStatsManager.close();
            streamStatsManager = null;
        }
    }
}

File: android/VideoView/app/src/main/java/io/streamroot/dna/videoview/PlayerActivity.kt
package io.streamroot.dna.videoview

import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.MediaController
import android.widget.Toast
import android.widget.VideoView
import io.streamroot.dna.core.StreamrootDNA
import io.streamroot.dna.utils.stats.StatsView
import io.streamroot.dna.utils.stats.StreamStatsManager

class PlayerActivity : AppCompatActivity() {
    private lateinit var videoView: VideoView
    private lateinit var streamrootDnaStatsView: StatsView

    private var mStreamrootKey: String? = null
    private var mStreamUrl: String? = null

    private var streamrootDNA: StreamrootDNA? = null
    private var streamStatsManager: StreamStatsManager? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_player)

        mStreamrootKey = intent.extras?.getString("streamrootKey")
        mStreamUrl = intent.extras?.getString("streamUrl")

        videoView = findViewById(R.id.videoView)
        streamrootDnaStatsView = findViewById(R.id.streamrootDnaStatsView)
    }

    override fun onResume() {
        val streamrootDNA = initStreamroot()

        videoView.setVideoPath(streamrootDNA?.manifestUrl.toString())
        videoView.setOnPreparedListener({ mp -> mp.setVolume(.5f, .5f) })
        videoView.setMediaController(MediaController(this))
        videoView.requestFocus()
        videoView.start()

        super.onResume()
    }

    override fun onPause() {
        videoView.pause()

        super.onPause()
    }

    override fun onDestroy() {
        videoView.stopPlayback()
        stopStreamroot()

        super.onDestroy()
    }

    private fun initStreamroot(): StreamrootDNA? {
        var mSdk: StreamrootDNA? = null
        try {
            mSdk = StreamrootDNA.newBuilder()
                    .context(applicationContext)
                    .streamrootKey(mStreamrootKey)
                    .interactor(VideoViewPlayerInteractor(videoView))
                    .start(Uri.parse(mStreamUrl))

            streamStatsManager = StreamStatsManager.newStatsManager(mSdk, streamrootDnaStatsView)
            streamStatsManager?.start()
        } catch (e: Exception) {
            Toast.makeText(applicationContext, e.message, Toast.LENGTH_LONG)
                    .show()
        }

        return mSdk
    }

    private fun stopStreamroot() {
        streamrootDNA?.destroy()
        streamrootDNA = null

        streamStatsManager?.stop()
        streamStatsManager = null
    }
}
